# Note: Name of file should match name of robot in URDF

# Robot Execution Speed
main_velocity_scaling_factor: 0.6 # moving between cartesian paths
approach_velocity_scaling_factor: 0.2
lift_velocity_scaling_factor: 0.1
retreat_velocity_scaling_factor: 0.6
calibration_velocity_scaling_factor: 0.3  #0.1

# Sleep timers
wait_before_grasp: 0.1
wait_after_grasp: 1.0

# Poses
start_pose: manipulator_home
right_arm_dropoff_pose: goal_bin_pose
left_arm_dropoff_pose: left_goal_bin_pose_DISABLED

# Cartesian path config
jump_threshold: 4

# Safety
collision_wall_safety_margin: 0.01 # 0.02

# Semantics
dual_arm: 0
right_hand_name: gripper
left_hand_name: gripper_DISABLED
right_arm_name: manipulator
left_arm_name: manipulator_DISABLED
both_arms_name: manipulator_DISABLED

# Calibration
# 0.029 m is the width of frame to gantry center, to be added to the z component
# 0.035 m is the width of the 8020 bar in front of shelf
# in-lab distance is 0.5+0.029-0.035 = 0.494
#world_to_shelf_transform: [0.489, -0.438, 0.002, 0.0, 0.0, 0.0]  # IDEAL
world_to_shelf_transform: [0.67, -0.453, -0.038, 0.0, 0.0, 0.0]  # After arm offset

# Adding 0.38 m offset on arm_mount_joint
# Actual 0.167 offset

# 0.52 m from shelf (large frame to leg) before moving shelf 


# Topics
joint_state_topic: /joint_states

# Goal bin - different for each robot
goal_bin_x: -0.35
goal_bin_y: 0.0635
goal_bin_z: 0.5
place_goal_down_distance_desired: 0.22
goal_bin_clearance: 0.1

# Side limits (walls)
left_wall_y: 0 # 0 means disabled
right_wall_y: 0 # 0 means disabled

# Top limit
ceiling_z: 2.4

# Perception Data
camera:
  # THIS IS USED TO OFFSET THE CAMERA TO THE CIRCLAR CALIBRATION LOCATION
  x_translation_from_bin: 0.55 # meters
  y_translation_from_bin: 0.0 # negative is right looking at shelf
  # THIS IS USED TO OFFSET THE GANTRY FROM THE BIN CENTROID
  z_translation_from_bin: -0.66 # must be -0.66 or less
  x_rotation_from_standard_grasp: 0 #-0.2  # radians
  y_rotation_from_standard_grasp: 0  # negative is torwards right of shelf
  z_rotation_from_standard_grasp: 0  # radians
  lift_distance: 0.05
  left_distance: 0.05
  camera_frame_display_scale: 0.0005
  left_camera_frame: /xtion_left_depth_frame
  right_camera_frame: /xtion_right_depth_frame

# Test data
test:
  test_joint_limit_joint: 1  # starts at 0, negative number means test all
  test_double: 0.2 # for anything

behavior:
  end_effector_enabled: true
  super_auto: false